src/pymatgen/analysis/interfaces/coherent_interfaces.py

```diff
@@ -101,21 +101,15 @@ def _find_matches(self) -> None:
         for match in self.zsl_matches:
             xform = get_2d_transform(film_vectors, match.film_vectors)
             strain, _rot = polar(xform)
-            assert_allclose(
```

Comment:
Similarly, `np.testing.assert_allclose` is not suitable for production code (~ 2.5x slower than `np.isclose`), though it provides more detailed debug info for testing.

Note the default tolerances are a bit stricter for `assert_allclose`, this commit also migrated the tolerances as is without changing in behaviour.
> testing.assert_allclose(actual, desired, rtol=1e-07, atol=0, equal_nan=True, err_msg='', verbose=True, *, strict=False)

> numpy.allclose(a, b, rtol=1e-05, atol=1e-08, equal_nan=False)

Test script:
```python
import numpy as np
from numpy.testing import assert_allclose
import timeit


def run_tests():
    # Array sizes to test
    sizes = [(10, 10), (100, 100), (1000, 1000), (10000, 10000)]

    rtol = 1e-5
    atol = 1e-8

    # Run tests for each size
    for size in sizes:
        arr1 = np.random.rand(*size)
        arr2 = np.copy(arr1)  # Exact copy of arr1

        assert_allclose(arr1, arr2, rtol=rtol, atol=atol)
        assert np.allclose(arr1, arr2, rtol=rtol, atol=atol)

        # Measure performance in milliseconds
        num_runs = 10
        assert_time = timeit.timeit(lambda: assert_allclose(arr1, arr2, rtol=rtol, atol=atol), number=num_runs) * 1000
        allclose_time = timeit.timeit(lambda: np.allclose(arr1, arr2, rtol=rtol, atol=atol), number=num_runs) * 1000

        print(f"\nPerformance results for array size {size} over {num_runs} runs:")
        print(f"assert_allclose time: {assert_time:.3f} ms")
        print(f"np.allclose time: {allclose_time:.3f} ms")
        print(f"Speedup: {assert_time / allclose_time:.2f}x")

run_tests()
```

I got:

```
Performance results for array size (10, 10) over 10 runs:
assert_allclose time: 0.308 ms
np.allclose time: 0.102 ms
Speedup: 3.03x

Performance results for array size (100, 100) over 10 runs:
assert_allclose time: 0.563 ms
np.allclose time: 0.244 ms
Speedup: 2.30x

Performance results for array size (1000, 1000) over 10 runs:
assert_allclose time: 55.887 ms
np.allclose time: 22.179 ms
Speedup: 2.52x

Performance results for array size (10000, 10000) over 10 runs:
assert_allclose time: 12374.578 ms
np.allclose time: 2916.093 ms
Speedup: 4.24x
```